### PR TITLE
fix (jkube-kit/config) : `docker.buildArg.*` properties not taken into account in OpenShift plugins

### LINKED ISSUE #2904: docker.buildArg.* properties not taken into account in OpenShift plugins
### Describe the bug

The JKube Openshift Maven plugin fails to recognize specified Docker build-arg when they are injected into a Dockerfile from Maven properties. This issue manifests specifically when using Maven properties and does not occur with the XML/Gradle DSL configuration since https://github.com/eclipse/jkube/pull/2895.
This issue occurs despite the JKube [documentation](https://eclipse.dev/jkube/docs/openshift-maven-plugin/#_build_args) confirming that build arguments can be specified either via the plugin configuration or as Maven properties.

### Eclipse JKube version

SNAPSHOT

### Component

OpenShift Maven Plugin

### Apache Maven version

None

### Gradle version

None

### Steps to reproduce

1. **Set Up the Dockerfile:**
   Place the following content in `src/main/docker/Dockerfile`:
   ```dockerfile
   ARG FILE_EXTENSION
   FROM docker.io/alpine:latest as builder
   RUN touch test.${FILE_EXTENSION}

   FROM docker.io/alpine:latest
   COPY --from=builder /test.txt /test.txt

   ENTRYPOINT ["ls"]
   ```

2. **Configure the JKube Plugin:**
   Insert the JKube Openshift Maven plugin configuration into your `pom.xml`:
   ```xml
   <plugin>
       <groupId>org.eclipse.jkube</groupId>
       <artifactId>openshift-maven-plugin</artifactId>
       <version>1.17-SNAPSHOT</version>
       <configuration>
         <buildStrategy>docker</buildStrategy>
         <images>
           <image>
             <name>helloworld:%l</name>
             <build>
               <contextDir>${project.basedir}/src/main/docker/</contextDir>
               <dockerFile>Dockerfile</dockerFile>
             </build>
           </image>
         </images>
       </configuration>
   </plugin>
   ```

3. **Define the Build Argument:**
   Define the build argument for the Docker image in your `pom.xml`:
   ```xml
   <properties>
       <docker.buildArg.FILE_EXTENSION>txt</docker.buildArg.FILE_EXTENSION>
   </properties>
   ```

4. **Execute the Build Command:**
   Execute the build using the following Maven command:
   ```
   mvn oc:build
   ```

5. **Error Observation:**
   The build process fails, yielding the following output:
   ```
   [INFO] oc: --> abd4c9e7124f
   [INFO] oc: [1/2] STEP 3/3: RUN touch test.${FILE_EXTENSION}
   [INFO] oc: --> ee9fcbe057df
   [INFO] oc: [2/2] STEP 1/5: FROM docker.io/alpine
   [INFO] oc: [2/2] STEP 2/5: COPY --from=builder test.txt test.txt
   [ERROR] oc: Failed to execute the build [Unable to build the image using the OpenShift build service]
   [ERROR] oc: Failed to tail build log: %s: java.lang.IllegalStateException: java.io.IOException: channel already closed with exception
   ```

### Expected behavior

The `docker.buildArg.FILE_EXTENSION` property should be properly processed, enabling the substitution of `${FILE_EXTENSION}` in the Dockerfile and ensuring the successful build of the Docker image with the specified configurations.

### Runtime

OpenShift

### Kubernetes API Server version

1.25.3

### Environment

Linux

### Eclipse JKube Logs

_No response_

### Sample Reproducer Project

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

- [x] Requires https://github.com/eclipse-jkube/jkube/pull/2994 to be merged first

Added call to resolve build args from all sources in OpenShiftBuildServiceUtils so that build args get resolved from all the available sources (compared to just ImageConfig)

Fixes #2904

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/2996 (_2024-04-29T17:58:00Z_)

:gear: [JKube E2E Tests (8883180384)](https://github.com/eclipse-jkube/ci/actions/runs/8883180384) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2996)](https://github.com/eclipse-jkube/ci/actions/runs/8883180384/job/24389489996)

File: jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/openshift/OpenShiftBuildServiceUtils.java [hunk 1]
```diff
@@ -133,7 +134,7 @@ protected static BuildStrategy createBuildStrategy(
               .withNamespace(StringUtils.isEmpty(fromNamespace) ? null : fromNamespace)
             .endFrom()
             .withEnv(checkForEnv(imageConfig))
-            .withBuildArgs(Optional.ofNullable(buildConfig.getArgs()).orElse(Collections.emptyMap()).entrySet().stream()
+            .withBuildArgs(Optional.of(mergeBuildArgs(imageConfig, jKubeServiceHub.getConfiguration())).orElse(Collections.emptyMap()).entrySet().stream()
```
Comment (sunix) [inline jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/openshift/OpenShiftBuildServiceUtils.java:137]: as far as I remember, mergeBuildArgs is also taking prop from the local docker config which is not applicable for openshift remote docker build.

Comment (rohanKanojia) [reply jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/openshift/OpenShiftBuildServiceUtils.java:137]: You're right. How shall we handle this situation? Maybe add a `boolean` parameter in `mergeBuildArgs` to check whether to read local docker config or not?

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**